src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -177,11 +177,13 @@ internal class GenericEqualityComparer<T>: EqualityComparer<T> where T: IEquatab
         }
 
         // Equals method for the comparer itself.
+        // If in the future this type is made sealed, change the is check to GetType() == obj.GetType().
         public override bool Equals(Object obj) =>
-            obj != null && GetType() == obj.GetType();
+            obj != null && obj is GenericEqualityComparer<T>;
```

Comment:
Oh wow, I had always thought that `object o = null; o is string` was true. In fact that was why I thought the original implementation used `(as ...) != null` instead of just a simple is check. I just tested it out in a console application though and it's actually false. I'll remove the null check.